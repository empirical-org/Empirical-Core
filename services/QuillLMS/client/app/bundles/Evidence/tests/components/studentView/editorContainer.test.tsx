import * as React from 'react'
import { shallow, mount, } from 'enzyme'
import toJson from 'enzyme-to-json'

import EditorContainer from '../../../components/studentView/editorContainer'

describe('EditorContainer component', () => {
  describe('when the activity has loaded', () => {
    const handleTextChange = () => {}
    const innerRef = () => {}
    const resetText = () => {}
    const stripHtml = (str) => { 'string' }
    const wrapper = mount(<EditorContainer
      className='step'
      disabled={false}
      handleTextChange={handleTextChange}
      html="Here is some text <u>because</u>&nbsp;"
      innerRef={innerRef}
      resetText={resetText}
      stripHtml={stripHtml}
      unsubmittableResponses={[]}
    />)

    it('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })

  })
})
