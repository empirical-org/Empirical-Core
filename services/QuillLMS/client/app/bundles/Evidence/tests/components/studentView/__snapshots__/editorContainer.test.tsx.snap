// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorContainer component when the activity has loaded renders 1`] = `
<EditorContainer
  className="step"
  disabled={false}
  handleTextChange={[Function]}
  html="Here is some text <u>because</u> "
  innerRef={[Function]}
  resetText={[Function]}
  stripHtml={[Function]}
  unsubmittableResponses={Array []}
>
  <div
    className="editor-container"
  >
    <ContentEditable
      className="step"
      data-gramm={false}
      disabled={false}
      html="Here is some text <u>because</u> "
      innerRef={[Function]}
      onChange={[Function]}
      spellCheck={false}
    >
      <div
        className="step"
        contentEditable={true}
        dangerouslySetInnerHTML={
          Object {
            "__html": "Here is some text <u>because</u> ",
          }
        }
        data-gramm={false}
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        spellCheck={false}
      />
    </ContentEditable>
    <button
      className="disabled clear-button"
      type="button"
    >
      <img
        alt="circle with an x in it"
        className="clear"
        src="undefined/images/icons/clear-disabled.svg"
      />
      Clear response
    </button>
  </div>
</EditorContainer>
`;
